src/core/server/http/http_server.ts

```diff
@@ -189,13 +194,13 @@ export class HttpServer {
   }
 
   public async stop() {
+    this.stopped = true;
     if (this.server === undefined) {
       return;
     }
 
     this.log.debug('stopping http server');
     await this.server.stop();
-    this.server = undefined;
   }
```

Comment:
Would it be too pragmatic to have `stop` await a promise that would be resolved at the end of `start`?
Are there practical use cases where `stop` **should** be called executed `setup/start` and complete (if they were invoked). 

In most graceful shutdown implementations I'm aware of, this is what is actually done.

```ts
setup() {
  this.startingPromise = new Promise();
  // do setup
}

start() {
   // do start
   this.startingPromise.resolve(); // yea I know, this doesn't work like that. You get the idea.
}

stop() {
   if(this.startingPromise) {
      await this.startingPromise()
   }
  // do stop
}
```

Reply:
Good idea, I will try that tomorrow.

Reply:
I suspect there will be a lot of complaints that Kibana doesn't react to the `stop` signal and continue working. But  👍 as a temporary solution.